base/abstractarraymath.jl

```diff
@@ -166,15 +166,7 @@ end
 
 ## Permute array dims ##
 
-function permutedims(B::AbstractArray, perm)
-    dimsB = size(B)
-    ndimsB = length(dimsB)
-    (ndimsB == length(perm) && isperm(perm)) || throw(ArgumentError("no valid permutation of dimensions"))
-    dimsP = ntuple(i->dimsB[perm[i]], ndimsB)::typeof(dimsB)
-    P = similar(B, dimsP)
-    permutedims!(P, B, perm)
-end
-
+permutedims(B::AbstractArray, perm) = permutedims(convert(Array, B), perm)
 
```

Comment:
This dispatches ultimately to `permutedims!(::Array, ::StridedArray, perm)`. It's on my TODO list to write a generic `permutedims!`, but in the meantime this seems to be the only safe way to make sure that it calls a valid method (we don't have a `similar_strided` function).